pinot-core/src/main/java/org/apache/pinot/core/transport/ServerChannels.java

```diff
@@ -94,9 +97,10 @@ public void shutDown() {
 
   @ThreadSafe
   private class ServerChannel {
-    final TSerializer _serializer = new TSerializer(new TCompactProtocol.Factory());
     final ServerRoutingInstance _serverRoutingInstance;
     final Bootstrap _bootstrap;
+    // lock to protect channel as requests must be written into channel sequentially
+    private final ReentrantLock _channelLock = new ReentrantLock();
```

Comment:
(minor) `private` is redundant as the whole class is private

Reply:
done